parsers/ada.c

```diff
@@ -1219,38 +1219,46 @@ static adaTokenInfo *adaParseType(adaTokenInfo *parent, adaKind kind)
       movePos(1);
       adaParseVariables(token, ADA_KIND_ENUM_LITERAL);
     }
-    else if(adaKeywordCmp(ADA_KEYWORD_RECORD))
-    {
-      /* until we hit "end record" we need to gather type variables */
-      while(exception != EXCEPTION_EOF)
-      {
-        skipWhiteSpace();
-
-        if(adaKeywordCmp(ADA_KEYWORD_END))
-        {
-          skipWhiteSpace();
-          if(adaKeywordCmp(ADA_KEYWORD_RECORD))
-          {
-            break;
-          }
-          skipPast(";");
-        } /* if(adaKeywordCmp(ADA_KEYWORD_END)) */
-        /* handle variant types */
-        else if(adaKeywordCmp(ADA_KEYWORD_CASE))
-        {
-          skipPastKeyword(ADA_KEYWORD_IS);
-        }
-        else if(adaKeywordCmp(ADA_KEYWORD_WHEN))
-        {
-          skipPast("=>");
-        }
-        else
-        {
-          adaParseVariables(token, ADA_KIND_RECORD_COMPONENT);
-          skipPast(";");
-        }
-      } /* while(true) - end of record not found */
-    } /* else if(adaKeywordCmp(ADA_KEYWORD_RECORD)) */
+	else
+	{
+		if(adaKeywordCmp(ADA_KEYWORD_NEW))
+		{
+			skipPastKeyword(ADA_KEYWORD_WITH);
```

Comment:
This doesn't work for input like:
```
procedure Derived_Types  is

    package Pak  is
       type Integer_1  is  range 1 .. 10;
       procedure P (I:  in Integer_1); -- primitive operation, assumes 1 .. 10
       type Integer_2  is  new Integer_1  range 8 .. 10; -- must not break P's assumption
```
Quoted from https://en.wikibooks.org/wiki/Ada_Programming/Type_System